c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1106 @@
+        char _dummy[2 * (sizeof(type_ident) == sizeof(uint)) - 1]; \
+    }
+
+// ---------------------------------------------------------------------
+// 1. Chip Organization
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 2. System Architecture
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 3. ARM968 Processing Subsystem
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 4. ARM 968
+
+// No special registers here
+
+// ---------------------------------------------------------------------
+// 5. Vectored Interrupt Controller
+
+typedef void (*vic_interrupt_handler_t) (void);
+
+typedef union {
+    struct {
+        uint watchdog : 1;
+        uint software : 1;
+        uint comm_rx : 1;
+        uint comm_tx : 1;
+        uint timer1 : 1;
+        uint timer2 : 1;
+        uint cc_rx_ready : 1;
+        uint cc_rx_parity_error : 1;
+        uint cc_rx_framing_error : 1;
+        uint cc_tx_full : 1;
+        uint cc_tx_overflow : 1;
+        uint cc_tx_empty : 1;
+        uint dma_done : 1;
+        uint dma_error : 1;
+        uint dma_timeout : 1;
+        uint router_diagnostic : 1;
+        uint router_dump : 1;
+        uint router_error : 1;
+        uint cpu : 1;
+        uint ethernet_tx : 1;
+        uint ethernet_rx : 1;
+        uint ethernet_phy : 1;
+        uint slow_clock : 1;
+        uint cc_tx_not_full : 1;
+        uint cc_rx_mc : 1;
+        uint cc_rx_p2p : 1;
+        uint cc_rx_nn : 1;
+        uint cc_rx_fr : 1;
+        uint int0 : 1;
+        uint int1 : 1;
+        uint gpio8 : 1;
+        uint gpio9 : 1;
+    };
+    uint value;
+} vic_mask_t;
+
+typedef struct {
+    const vic_mask_t irq_status;
+    const vic_mask_t fiq_status;
+    const vic_mask_t raw_status;
+    vic_mask_t int_select;
+    vic_mask_t int_enable;
+    vic_mask_t int_disable;
+    vic_mask_t soft_int_enable;
+    vic_mask_t soft_int_disable;
+    bool protection;
+    const uint _padding[3];
+    vic_interrupt_handler_t vector_address;
+    vic_interrupt_handler_t default_vector_address;
+} vic_control_t;
+
+typedef struct {
+    uint source : 5;
+    uint enable : 1;
+    uint : 26;
+} vic_vector_control_t;
+
+ASSERT_WORD_SIZED(vic_mask_t);
+ASSERT_WORD_SIZED(vic_interrupt_handler_t);
+ASSERT_WORD_SIZED(vic_vector_control_t);
+
+// ---------------------------------------------------------------------
+// 6. Counter/Timer
+
+typedef struct {
+    uint one_shot : 1;
+    uint size : 1;
+    uint pre_divide : 2;
+    uint : 1;
+    uint interrupt_enable : 1;
+    uint periodic_mode : 1;
+    uint enable : 1;
+    uint : 24;
+} timer_control_t;
+
+enum timer_pre_divide {
+    TIMER_PRE_DIVIDE_1 = 0,
+    TIMER_PRE_DIVIDE_16 = 1,
+    TIMER_PRE_DIVIDE_256 = 2
+};
+
+typedef struct {
+    uint status : 1;
+    uint : 31;
+} timer_interrupt_status_t;
+
+typedef struct {
```

Comment:
comment

Reply:
can i propose that all structs are named at top and bottom./ for readibility purposes?

Reply:
documented

Reply:
Propose away. No.